### DISCUSSION TITLE
Changes in `.github/workflows/` silently enable Actions on a forked repo

### DISCUSSION THREAD

**Discussion by tchebb** [community/community#53510] [Actions]
### Select Topic Area

Bug

### Body

As described in github/docs#15761, workflows won't run on a newly-created fork until the fork owner goes to the "Actions" tab and dismisses the "Workflows aren’t being run on this forked repository" interstitial.

But it's not quite as simple as that: yesterday, I pushed several feature branches to a fork of mine and, to my surprise, workflows ran for two of those branches but not the third. While investigating why, I learned about the interstitial above and saw that it wasn't present on my fork even though I'm 100% sure I never saw it (or dismissed it) on any repo prior to today.

After a fair bit of testing on other forks, I discovered that the interstitial will be automatically and silently dismissed, enabling Actions, whenever all of the following happen:
 - A branch is created or updated in the fork (has to be a branch, not a tag)
 - The new commits in that branch, in aggregate, do *at least one* of the following:
   - Add a new workflow file in `.github/workflows/`
   - For *every* existing workflow in `.github/workflows/`, either edit that workflow's steps (editing name and/or whitespace isn't enough) or delete it

Note that this happens even when the branch update is caused by clicking "Sync fork > Update branch", assuming the upstream branch has new commits matching the criteria above.

This seems like a bug to me, or at least something that should be documented: if I've explicitly chosen not to enable Actions on a fork, I don't want changes from upstream enabling them for me. Plus, it adds a lot of confusion for users who never see the interstitial but see its effects for pushes they make before unknowingly making one that dismisses it.

**Comment by jsoref** [comment]
Hmm, based on my note near the bottom of that issue, it's possible that the behavior has changed. I've definitely seen (and complained about somewhere) the behavior where pulling in things will enable actions...

There are so many rough edges relating to disabled workflows.
